site/src/pages/release-notes/1.10.0.mdx

```diff
@@ -0,0 +1,286 @@
+  HttpResponse recovered = 
+      response.recover(cause -> HttpResponse.of("Fallback"));
+  assert recovered.aggregate().join().contentUtf8().equals("Fallback");
+  ```
+- You can now resume a failed <type://StreamMessage> with a fallback <type://StreamMessage>. #3674
+  ```java
+  DefaultStreamMessage<Integer> stream = new DefaultStreamMessage<>();
+  stream.write(1);
+  stream.write(2);
+  stream.close(new IllegalStateException("Oops...");
+  StreamMessage<Integer> resumed = 
+      stream.recoverAndResume(cause -> StreamMessage.of(3, 4));
+  assert resumed.collect().join().equals(List.of(1, 2, 3, 4));
+  ```
+- You can now transform an error of <type://StreamMessage> into another using a function. #3668
+  ```java
+  StreamMessage stream = 
+      StreamMessage.aborted(ClosedStreamException.get());
+  StreamMessage transformed = stream.mapError(ex -> {
+    if (ex instanceof ClosedStreamException) {
+      return new IllegalStateException(ex);
+    } else {
+      return ex;
+    }
+  }); 
+  ```
+- You can now automatically encode an object into JSON for a response payload using
+  <type://HttpResponse#ofJson(Object)>. #3662
+  ```java
+  MyObject myObject = ...;
+  HttpResponse.ofJson(myObject);
+  MyError myError = ...;
+  HttpResponse.ofJson(HttpStatus.INTERNAL_SERVER_ERROR, myError);
+  ```
+- You can now fluently inject `io.grpc.ServerInterceptor`s using
+  <type://GrpcServiceBuilder#intercept(Iterable)>.
+  ```java
+  GrpcService.builder()
+             .addService(myService)
+             .intercept(myInterceptror)
+             .build();
+  ``` 
+- You can now easily add `Accept` headers with <type://RequestHeadersBuilder#accept(Iterable)>. #3704
+  ```java
+  RequestHeaders.builder()
+                .accept(MediaType.JSON)
+                .accept(MediaType.PLAIN_TEXT);
+  ```
+- You can now initiate graceful connection shutdown using
+  <type://ServiceRequestContext#initiateConnectionShutdown()>. #3516 #3715
+- You can now specify the default `ObjectMapper` using <type://JacksonObjectMapperProvider>. #3728
+  ```java
+  public class MyObjectMapperProvider 
+      implements JacksonObjectMapperProvider {
+  
+    @Override
+    public ObjectMapper newObjectMapper() {
+      return JsonMapper
+         .builder()
+         .addModules(new KotlinModule())
+         .enable(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES)
+         .build();
+    }
+  }
+  ```
+- You can now serve `application/grpc-web+proto` and `application/grpc-web-text+proto`
+  [protocols](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2)
+  using <type://AbstractUnaryGrpcService>. #3638 #3716
+- gRPC trailers are available in the <type://RequestContext> for unary gRPC server and client. #3724 #3739
+  ```java
+  UnaryGrpcClient client = ...;
+  try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+      client.execute("/com.example.MyService/UnaryCall", request); 
```

Comment:
```suggestion
      client.execute("/com.example.MyService/UnaryCall", request).aggregate(); 
```
to prevent the leak?

Reply:
It is `UnaryGrpcClient`, not `WebClient`. 😆

Reply:
Oops. 😅